### PR TITLE
Support host rewrite

### PR DESCRIPTION
**Pull request by luizbafilho**
Adds the option of host rewrite by using the kubernetes.io/ingress.host-rewrite annotation in the ingress.

closes: #69 

Signed-off-by: Luiz Filho <luizbafilho@gmail.com>

### CODE REVIEW

**Comment by davecheney** [comment]
@luizbafilho thank you for this PR. As per the contibuting guidelines I would prefer to see a discussion of the change you plan to make before you send code. In the future, please discussion what you plan to do before sending a PR.

File: internal/contour/translator_test.go [hunk 1]
```diff
@@ -462,7 +462,7 @@ func TestTranslatorAddIngress(t *testing.T) {
 			Domains: []string{"*"},
 			Routes: []*v2.Route{{
 				Match:  prefixmatch("/"),
-				Action: clusteraction("default/backend/80"),
+				Action: clusteraction("default/backend/80", map[string]string{}),
```
Comment (davecheney) [inline internal/contour/translator_test.go]: you can write `nil` here and everywhere else in place of `map...`, however I'd prefer it you added a new helper, say `clusteractionrewrite` that took the second map parameter.

**Comment by davecheney** [comment]
@luizbafilho i am making this PR closed until we can discussion the design of the `kubernetes.io/ingress.host-rewrite` annotation. For example, what are it's parameters, how it is validated, how does it apply to the default ingress, etc. Please feel free to reopen this PR once the discussion on the implementation on #69 is complete.

**Comment by luizbafilho** [comment]
@davecheney sorry about that, I might have missed the discussion part of the guideline. 

I would like to ask you to drive the discussion.I don't know your requirements. Can you list there what are your questions? so I can try to respond. 

I do open source on GitHub, but the projects were always mine. I'm not sure what I'm supposed to ask you before implementing a feature. 

I would like you to guide me on this. I finally found a project I like and I'm willing to become a frequent contributor.

**Comment by davecheney** [comment]
@luizbafilho thanks. I recommend looking at the tags for issues marked help wanted or good first issue.

**Pull request closed without merging**